modules/core/src/stat.cpp

```diff
@@ -4053,6 +4053,27 @@ int normHamming(const uchar* a, int n)
         result = vgetq_lane_s32 (vreinterpretq_s32_u64(bitSet2),0);
         result += vgetq_lane_s32 (vreinterpretq_s32_u64(bitSet2),2);
     }
+#elif CV_AVX2
+    {
+        __m256i _r0 = _mm256_setzero_si256();
+        __m256i _0 = _mm256_setzero_si256();
+        __m256i _popcnt_table = _mm256_setr_epi8(0, 1, 1, 2, 1, 2, 2, 3, 1, 2, 2, 3, 2, 3, 3, 4,
+                                                 0, 1, 1, 2, 1, 2, 2, 3, 1, 2, 2, 3, 2, 3, 3, 4);
+        __m256i _popcnt_mask = _mm256_set1_epi8(0x0F);
+
+        for(; i <= n - 32; i+= 32)
+        {
+            __m256i _a0 = _mm256_loadu_si256((const __m256i*)(a + i));
+
+            __m256i _popc0 = _mm256_shuffle_epi8(_popcnt_table, _mm256_and_si256(_a0, _popcnt_mask));
+            __m256i _popc1 = _mm256_shuffle_epi8(_popcnt_table,
+                             _mm256_and_si256(_mm256_srli_epi16(_a0, 4), _popcnt_mask));
+
+            _r0 = _mm256_add_epi32(_r0, _mm256_sad_epu8(_0, _mm256_add_epi8(_popc0, _popc1)));
+        }
+        _r0 = _mm256_add_epi32(_r0, _mm256_shuffle_epi32(_r0, 2));
+        result = _mm256_extract_epi32(_mm256_add_epi32(_r0, _mm256_permute2x128_si256(_r0, _r0, 1)), 0);
```

Comment:
`_mm256_extract_epi32` is not available in MSVS 2015.
I believe because it is not real intrinsic (there is no CPU command for this).

1) Looks like this works with MSVS (only under `#ifdef _MSC_VER` guard):
```
result = _mm256_add_epi32(_r0, _mm256_permute2x128_si256(_r0, _r0, 1)).m256i_i32[0];
```
2) Split into 2 intrinsic instructions (should be cross-platform):
```
__m128i vres128 = _mm256_extracti128_si256(_mm256_add_epi32(_r0, _mm256_permute2x128_si256(_r0, _r0, 1)),0);
result = _mm_extract_epi32(vres128,0);
```